### PR TITLE
Add libgit2/1.0.1

### PR DESCRIPTION
**Pull request by intelligide**
Specify library name and version:  **libgit2/1.0.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libgit2/all/conandata.yml [hunk 1]
```diff
@@ -1,7 +1,4 @@
 sources:
-  "0.28.5":
```
Comment (SSE4) [inline recipes/libgit2/all/conandata.yml:2]: please don't remove old versions

**Review by uilianries — CHANGES_REQUESTED** [review]
dont remove old versions

**Comment by intelligide** [comment]
> dont remove old versions

https://github.com/conan-io/conan-center-index/pull/3070#discussion_r498409627 ;)

**Review by uilianries — APPROVED** [review]

**Comment by uilianries** [comment]
@intelligide Thanks! It was a mess!

**Review by theirix — APPROVED** [review]
Looks good

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**